transport/x/cherami/transport.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cherami
+
+import (
+	"fmt"
+
+	"go.uber.org/yarpc/api/transport"
+	intsync "go.uber.org/yarpc/internal/sync"
+	"go.uber.org/yarpc/transport/x/cherami/internal"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/uber/cherami-client-go/client/cherami"
+)
+
+// TransportConfig defines the config in order to create a cherami transport
+// ServiceName should be the name of the service that is using yarpc
+// if HyperbahnHostFile is provided, hyperbahn will be used to connect to cherami
+// Otherwise, the provided frontend IP and port will be used to connect to cherami
+type TransportConfig struct {
+	ServiceName       string
+	HyperbahnHostFile string
+	FrontendIP        string
+	Port              int
+}
+
+// NewTransport creates a new cherami transport for shared objects between inbound and outbound
+func NewTransport(config TransportConfig) *Transport {
```

Comment:
@datoug I was expecting `cherami.NewTransport`to take a Cherami client via `TransportConfig`...

Let's jam tomorrow morning so I can understand why the client (client factory here?) should be created inside of this function instead of being injected. Cool?